### PR TITLE
Podman-friendly CONTAINER_ENGINE setting

### PR DESCRIPTION
**Pull request by dmitris**
PR changes the setting of `CONTAINER_ENGINE` in tests/vmtests/fetch-data.sh to allow overwriting the variable with for example `sudo podman` while keeping `docker` as the default.  It also adds a link to the vmtests README in the contributor guide.  

Based on the Slack conversation https://cilium.slack.com/archives/C2B917YHE/p1660856168376349?thread_ts=1660843012.876959&cid=C2B917YHE - thanks @kkourt!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,8 @@
 lib/*
 /tetragon
 /tetragon-alignchecker
+/tetragon-tester
+/tetragon-vmtests-run
```
Comment (kkourt) [inline .gitignore:6]: Why is this needed? Typically, these utilities are build iin tests/vmtests.

See: https://github.com/cilium/tetragon/blob/main/tests/vmtests/.gitignore

Comment (dmitris) [reply .gitignore]: removed, let me know if I should rebase / force-push

**Review by kkourt — APPROVED** [review]
Thanks!

**Comment by dmitris** [comment]
@kkourt could you please restart the failed build? The build failure seems not related to this change.

**Review by kkourt — APPROVED** [review]
